aprilgroup_tracking/main.py

```diff
@@ -0,0 +1,41 @@
+'''
+The main.py will call all the necessary functionality for this project.
+So far it calibrates the camera, detects the apriltags and estimates the pose of the aprilgroup, drawing, it onto
+an OpenCV window.
+'''
+import os
+from logging_results.create_logs import CustomLogger
+from calibration.calibrate_camera import Calibration
+
+
+def main():
+
+    # Create a folder called "logs"
+    log_directory = "logs"
+    try:
+        if not os.path.exists(log_directory):
+            os.makedir(log_directory)
+    except IsADirectoryError:
+        raise ValueError("Could not create log directory")
+
+    # Calibration logs
+    calibration_logger = CustomLogger(log_file=log_directory+"/camera_calibration_logs", name ="camera_calibration_logs")
+    # Pose Estimation Logs
+    det_pose_logger = CustomLogger(log_file=log_directory+"/detection_pose_estimation_logs", name ="detection_pose_estimation_logs")
+
+    # If Camera Parameters already exists, load them, else calculate them by calibrating the camera.
+    calibrate = Calibration(calibration_logger)
+    if calibrate.try_load_intrinsic():
+        mtx, dist, rvecs, tvecs = calibrate.mtx, calibrate.dist, calibrate.rvecs, calibrate.tvecs
+        print(mtx)
+    else:
+        calibrate.start_intrinsic_calibration()
```

Comment:
Will calling start_intrinsics_calibration() without following up with calculate_intrinsics() immediately after ever be performed? If not, I suggest calling start_intrinsics_calibration()  at the beginning of calculate_intrinsics().
Also, perhaps start_intrinsics_calibration() should return objp as it's not used beyond calculate_intrinsics() (so no need for it to be an instance variable)